runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -1404,3 +1404,45 @@ TR_JITServerRelocationRuntime::copyDataToCodeCache(const void *startAddress, siz
    return coldCodeStart;
    }
 #endif /* defined(J9VM_OPT_JITSERVER) */
+
+void
+printAOTHeader(const void* aotHeaderAddress, char * buff, const size_t BUFF_SIZE)
+   {
+   memset(buff, 0, BUFF_SIZE*sizeof(char));
+   if (!aotHeaderAddress)
+      {
+      strncat(buff, "null", BUFF_SIZE - strlen(buff) - 1);
+      return;
+      }
+
+   PORT_ACCESS_FROM_PORT(TR::Compiler->portLib);
+   OMRPORT_ACCESS_FROM_OMRPORT(TR::Compiler->omrPortLib);
+   TR_AOTHeader * hdrInCache = (TR_AOTHeader *)aotHeaderAddress;
```

Comment:
Is it possible do change the first parameter to `TR_AOTHeader *aotHeaderAddress` to avoid this cast. You would need to predeclare `struct TR_AOTHeader` in the file where `printAOTHeader` is used.

Reply:
Might not be possible since `printAOTHeader` is called from the VM.

Reply:
I think it's possible, with forward declaration. Not sure that's cleaner than this though

Reply:
change added here: ddd80a3b38b8e2ea9e0537d87798ab4824ece58f